Lesson35/src/modules/user/Account/index.jsx

```diff
@@ -0,0 +1,24 @@
+import { UserDetailsForm } from '../UserDetailsForm';
+import { useContext, useState } from 'react';
+import { UserContext, UserDispatchContext } from '../UserProvider';
+
+export const Account = () => {
+  const user = useContext(UserContext);
+  const dispatch = useContext(UserDispatchContext);
+  const [showDetails, setShowDetails] = useState(true);
+
+  const handleNameChange = (newName) => {
+    dispatch({ type: 'UPDATE_NAME', payload: newName });
+  };
+
+  return (
+    <div>
+      {showDetails && user && (
+        <>
+          <h1>Welcome, {user.name || "Guest"}</h1>
+          <UserDetailsForm username={user.name} onNameChange={handleNameChange} />
```

Comment:
Don't pass the dispatch function to the UserDetailsForm. Instead, add `  const dispatch = useContext(UserDispatchContext);` to UserDetailsForm component directly and handle update from there.


It's ok to pass username as prop since it's only passed to the child component and used both in the Account and UserDetailsForm components.